### PR TITLE
Feature: Studio

### PR DESCRIPTION
**Pull request by kiwicopple**
## What kind of change does this PR introduce?

Starts the work on a self-hosted dashboard.

- [context](https://github.com/supabase/supabase/discussions/1001#discussioncomment-558696)
- Livestream
  - [Build in Public 006: Supabase Studio](https://www.youtube.com/watch?v=LHYrqBb4q9I)
  - [Build in Public 007: Supabase Studio - API Docs](https://www.youtube.com/watch?v=R4gJhX_JFTo)
  - [Build in Public 008: Supabase Studio](https://www.youtube.com/watch?v=x5TKDQ-w3OQ)
  - [Build in Public 009: Supabase Studio - Postgres Meta](https://www.youtube.com/watch?v=wNoVi3wAmh0)

### CODE REVIEW

**Comment by Rohithgilla12** [comment]
Hey, @kiwicopple added #2309 fixing `required` not showing bug and added `description` to the table.

File: studio/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    "noEmit": true,
+    "esModuleInterop": true,
+    "module": "esnext",
+    "moduleResolution": "node",
+    "resolveJsonModule": true,
+    "isolatedModules": true,
+    "jsx": "preserve",
+    "baseUrl": "."
+  },
+  "include": [
```
Comment (KoltonG) [inline studio/tsconfig.json:22]: Thoughts on using TS paths to reduce the `../../components` imports?

```json
"paths": {
    "@components": ["components"],
}
```

File: studio/pages/docs/tables/[tableName].js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+import DocsLayout from '../../../components/layouts/DocsLayout'
+import Loading from '../../../components/utils/Loading'
+import Error from '../../../components/utils/Error'
+import { fetchOpenApiSpec } from '../../../lib/api'
+import { useRouter } from 'next/router'
+import { Typography } from '@supabase/ui'
+
+export default function Home() {
```
Comment (KoltonG) [inline studio/pages/docs/tables/[tableName].js:8]: Curious as to why you've went with `.js` for the `auth/` and `docs/` folders?

File: studio/pages/_app.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import '../styles/globals.css'
+import type { AppProps } from 'next/app'
+
+
```
Comment (KoltonG) [inline studio/pages/_app.tsx:4]:
```suggestion
```

File: studio/styles/globals.css [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
+}
+
+a {
+  color: inherit;
+  text-decoration: none;
+}
+
+* {
+  box-sizing: border-box;
```
Comment (KoltonG) [inline studio/styles/globals.css:19]: I find it interesting that @tailwind base doesn't include this 🤔

File: studio/public/vercel.svg [hunk 5]
```diff
@@ -0,0 +1,4 @@
+<svg width="283" height="64" viewBox="0 0 283 64" fill="none"
```
Comment (KoltonG) [inline studio/public/vercel.svg:1]: Wondering if you are still using this?

File: studio/components/layouts/DashboardLayout.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+const SidebarMenu = ({ sidebar }: { sidebar: Sidebar }) => {
+  return (
+    <div
+      className={[
+        'w-64 h-screen overflow-autohide-scrollbar border-r ', // Layout
+        'bg-sidebar-linkbar-light', // Light mode
+        'dark:bg-sidebar-linkbar-dark dark:border-dark ', // Dark mode
+      ].join(' ')}
```
Comment (KoltonG) [inline studio/components/layouts/DashboardLayout.tsx:54]: Ahh this is fun!

File: studio/pages/auth/settings.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import AuthLayout from '../../components/layouts/AuthLayout'
+
+export default function Home() {
+
```
Comment (KoltonG) [inline studio/pages/auth/settings.js:4]:
```suggestion
```

File: studio/pages/table.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import Head from 'next/head'
+import DashboardLayout from '../components/layouts/DashboardLayout'
+import styles from '../styles/Home.module.css'
```
Comment (KoltonG) [inline studio/pages/table.tsx:3]: It doesn't seem like this file exist anymore 😢

**Review by KoltonG — APPROVED** [review]
Love it! I'll have to take a look at the WIP videos 😄

File: studio/pages/docs/tables/[tableName].js (hunk 2 above)
Comment (i-pip) [reply studio/pages/docs/tables/[tableName].js:8]: 💯 With you 100% on this, I just switched all these components to `.tsx` and already prevented a few potential runtime bugs popping up.

File: studio/tsconfig.json (hunk 1 above)
Comment (kiwicopple) [reply studio/tsconfig.json:22]: Next.js actually supports this out-of-the-box with tsconfig: https://nextjs.org/docs/advanced-features/module-path-aliases

we won't need to add `@components` - we can just write `components/XXX` and it should work 👍

File: studio/pages/docs/tables/[tableName].js (hunk 2 above)
Comment (kiwicopple) [reply studio/pages/docs/tables/[tableName].js]: fixed by @i-pip ⚡

File: studio/components/layouts/DashboardLayout.tsx (hunk 6 above)
Comment (kiwicopple) [reply studio/components/layouts/DashboardLayout.tsx:61]: that was for my own sanity 😅  Perhaps will need to a refactor, but works for now!

File: studio/tsconfig.json (hunk 1 above)
Comment (KoltonG) [reply studio/tsconfig.json:22]: Roger! Yeah, the baseUrl, in this case, would solve that!

File: studio/store/postgres/MetaStore.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+   */
+  async query(query) {
+    // let headers = {}
+    // // headers['Content-Type'] = 'application/json'
+    // const url = `${this.baseUrl}/query`
+    // const { data} = await post(url, payload).ca
+    // if (data.error) {
+    //   throw data.error
+    // }
+    // return data
```
Comment (itsalb3rt) [inline studio/store/postgres/MetaStore.js:45]: I think this commented code should not be here.

File: studio/README.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+Run this repo
+
+```bash
+# Add your API URL and KEY as environment variables
+# You can find you API params at app.supabase.io > project > settings > API
+
+NEXT_PUBLIC_SUPABASE_URL=URL
+NEXT_PUBLIC_SUAPBASE_ANON_KEY=API_KEY
```
Comment (DiegoBM) [inline studio/README.md:40]: Is this a typo? _SUAPBASE_

Comment (w3b6x9) [reply studio/README.md:40]: i think so

Comment (MildTomato) [reply studio/README.md:40]: hahah spent 5 mins trying to find out why the env var wasn't working :joy:

**Review by serebano — APPROVED** [review]

**Review by itsalb3rt — APPROVED** [review]

**Pull request closed without merging**